receiver/awscontainerinsightreceiver/internal/stores/podstore.go

```diff
@@ -467,45 +474,80 @@ func (p *PodStore) addStatus(metric CIMetric, pod *corev1.Pod) {
-			for _, containerStatus := range pod.Status.ContainerStatuses {
-				if containerStatus.Name == containerName {
+			for _, status := range pod.Status.ContainerStatuses {
+				if status.Name == containerName {
+					var cs containerStatus
 					switch {
-					case containerStatus.State.Running != nil:
+					case status.State.Running != nil:
 						metric.AddTag(ci.ContainerStatus, "Running")
-					case containerStatus.State.Waiting != nil:
+						cs.running = true
+					case status.State.Waiting != nil:
 						metric.AddTag(ci.ContainerStatus, "Waiting")
-						if containerStatus.State.Waiting.Reason != "" {
-							metric.AddTag(ci.ContainerStatusReason, containerStatus.State.Waiting.Reason)
+						cs.waiting = true
+						if status.State.Waiting.Reason != "" {
+							metric.AddTag(ci.ContainerStatusReason, status.State.Waiting.Reason)
+							if strings.Contains(status.State.Waiting.Reason, "Crash") {
+								cs.waitingCrashed = true
+							}
 						}
-					case containerStatus.State.Terminated != nil:
+					case status.State.Terminated != nil:
 						metric.AddTag(ci.ContainerStatus, "Terminated")
-						if containerStatus.State.Terminated.Reason != "" {
-							metric.AddTag(ci.ContainerStatusReason, containerStatus.State.Terminated.Reason)
+						cs.terminated = true
+						if status.State.Terminated.Reason != "" {
+							metric.AddTag(ci.ContainerStatusReason, status.State.Terminated.Reason)
 						}
 					}
 
-					if containerStatus.LastTerminationState.Terminated != nil && containerStatus.LastTerminationState.Terminated.Reason != "" {
-						metric.AddTag(ci.ContainerLastTerminationReason, containerStatus.LastTerminationState.Terminated.Reason)
+					if status.LastTerminationState.Terminated != nil && status.LastTerminationState.Terminated.Reason != "" {
+						metric.AddTag(ci.ContainerLastTerminationReason, status.LastTerminationState.Terminated.Reason)
 					}
 					containerKey := createContainerKeyFromMetric(metric)
 					if containerKey != "" {
 						content, ok := p.getPrevMeasurement(ci.TypeContainer, containerKey)
 						if ok {
 							prevMeasurement := content.(prevContainerMeasurement)
 							result := 0
-							if int(containerStatus.RestartCount) > prevMeasurement.restarts {
-								result = int(containerStatus.RestartCount) - prevMeasurement.restarts
+							if int(status.RestartCount) > prevMeasurement.restarts {
+								result = int(status.RestartCount) - prevMeasurement.restarts
 							}
 							metric.AddField(ci.ContainerRestartCount, result)
 						}
-						p.setPrevMeasurement(ci.TypeContainer, containerKey, prevContainerMeasurement{restarts: int(containerStatus.RestartCount)})
+						p.setPrevMeasurement(ci.TypeContainer, containerKey, prevContainerMeasurement{restarts: int(status.RestartCount)})
 					}
+
+					addContainerStatuses(metric, cs)
 				}
 			}
 		}
 	}
 }
 
+func addContainerStatuses(metric CIMetric, s containerStatus) {
+	if s.running {
```

Comment:
nit / style / non-blocking.  Could we have made a map with the key being the string (i.e. running/waiting) and the value being a bool? And then loop through the map and assign the values

Reply:
Oh, okay here's the problem now that I am implementing - if it's a generic map, I still have to look up each individual metric, or propagate this if-else branching for populating the map itself. If I only insert the running status when a container is running, I'm not going to get the other waiting and terminated statuses in the map

Reply:
I was relying on the default bool value in the struct to fall back on, so I didn't need to assign each value in the struct in the switch-case where I switch on the statuses

Reply:
Updated